README.md

```diff
@@ -1,81 +1,94 @@
+| Serializer | Jackson | Circe | Protobuf v3 | Avro | Borer | Kryo
+|:---|:---|:---|:---|:---|:---|:---|
+| Data formats | JSON or [CBOR](https://cbor.io) | JSON | JSON or binary | JSON or binary | JSON or CBOR | binary
+| Scala support | with [`jackson-module-scala`](https://github.com/FasterXML/jackson-module-scala) <ul><li>lacks support of basic scala types like `Unit`</li><li>without explicit annotation doesn't work with generics extending `AnyVal`</ul> | perfect | with [`ScalaPB`](https://scalapb.github.io) generates Scala ADTs based on protobuf definitions | with [`Avro4s`](https://github.com/sksamuel/avro4s) library generaters Avro schema based on Scala ADTs | perfect | perfect
+| Akka support | [library](https://doc.akka.io/docs/akka/current/serialization-jackson.html) | requires custom serializer | [akka.remote](https://doc.akka.io/docs/akka/current/serialization.html) | requires custom serializer | requires custom serializer | [library](https://github.com/altoo-ag/akka-kryo-serialization)
+| Runtime safety | none, uses reflection | encoders and decoders are checked during compile time | standard for scala code | standard for scala code | encoders and decoders are checked during compile time | none, uses reflections
+| Schema evolution | <ul><li>removing field</li><li>adding optional field</li></ul> with [`JacksonMigration`](https://doc.akka.io/docs/akka/current/serialization-jackson.html#schema-evolution) <ul><li>adding mandatory field</li><li>renaming field</li><li>renaming class</li><li>support of forward versioning for rolling updates</li></ul>| <ul><li>adding optional field</li><li>removing optional field</li><li>adding required field with default value</li><li>removing required field</li><li>renaming field</li><li>reordering fields</li><li>transforming data before deserialization</li></ul> | <ul><li>adding optional field</li><li>removing optional field</li><li>adding required field with default value</li><li>removing required field</li><li>renaming field</li><li>reordering fields</li><li>changing between compatible types</li></ul>  | <ul><li>reordering fields</li><li>renaming fields</li><li>adding optional field</li><li>adding required field with default value</li><li>removing field with default value</li></ul> | <ul><li>renaming fields</li><li>transforming data before deserialization</li></ul> | <ul><li>adding field</li><li>removing field</li><li>renaming field</li></ul>
+| Boilerplate | a lot: <ul><li>ADTs requires amount of annotation equal to or exceeding the actual type definitions</li><li>poor support for case object - requires custom serializer and deserializer and sealed trait per case object, otherwise, it creates new singletons and breaks pattern matching</ul> | a bit: every top level sealed trait must be mentioned for Circe to generate a codec | in case of custom types a second layer of models is needed | sometimes requires annotations | considerable: every top level sealed trait must have manually defined codec | none
```

Comment:
Hmmm Circe _a bit_ but Borer _considerable_... but the actual description looks almost identical 🤔

Reply:
I suppose the effort is considerable for Borer b/c you need to define codecs for transitively included types, while Circe relies on Magnolia for that, right?

Reply:
Yes, Circe relies on Magnolia.
Also Borer has separate codec[A] definition and register[A] call.